### ISSUE TITLE
GECKO Emulator Training

### ISSUE THREAD

**Issue by djgagne** [NCAR/gecko-ml#1]
I have started a skeleton of a framework for the GECKO emulator training pipeline. Prior to this, all training and evaluation has been done in Jupyter notebooks, but now we need to scale up to handle different datasets and to enable automatic hyperparameter searches and more systematic evaluation practices. To start this, I moved some code from the mlmicrophysics repository over here that could be useful.

The training pipeline should contain the following elements:
1. Read a yaml config file to get information about data and model configurations.
2. Load data and split it into training, validation and testing sets. For GECKO, the sets should be split by experiment.
3. Perform rescaling of the input and output values.
4. Evaluate whether the ML model should predict tendencies or the value at the next timestep. 
5. Train multiple ML model configurations in parallel.
6. Calculate evaluation metrics. We should look at metrics across the different species (precursor, gas, and aerosols).
7. Save model weights and data transformation parameters along with metrics, predictions and plots.

Please add more details and discuss implementation in this thread.
